File: backend/src/scans/dtos/scan-response.dto.ts
import { ApiProperty } from '@nestjs/swagger';

class ScanImageDto {
  @ApiProperty()
  id: number;

  @ApiProperty()
  path: string;

  constructor(image: { id: number; path: string }) {
    this.id = image.id;
    this.path = image.path;
  }
}

export class ScanResponseDto {
  @ApiProperty()
  id: number;

  @ApiProperty()
  patientUserId: number;

  @ApiProperty({ nullable: true })
  appointmentId: number | null;

  @ApiProperty({ type: [ScanImageDto] })
  images: ScanImageDto[];

  @ApiProperty()
  createdAt: Date;

  constructor(scan: {
    id: number;
    patientUserId: number;
    appointmentId: number | null;
    images?: { id: number; path: string }[];
    createdAt: Date;
  }) {
    this.id = scan.id;
    this.patientUserId = scan.patientUserId;
    this.appointmentId = scan.appointmentId;
    this.images = (scan.images || []).map((img) => new ScanImageDto(img));
    this.createdAt = scan.createdAt;
  }
}

File: mobile/MedAI/composeApp/src/commonMain/kotlin/org/example/project/data/remote/dto/scans/ScanResponseDto.kt
package org.example.project.data.remote.dto.scans

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class ScanResponseDto(
    @SerialName("id") val id: Int,
    @SerialName("patientUserId") val patientUserId: Int,
    @SerialName("appointmentId") val appointmentId: Int?,
    @SerialName("images") val images: List<ScanImageDto>,
    @SerialName("createdAt") val createdAt: String
)

@Serializable
data class ScanImageDto(
    @SerialName("id") val id: Int,
    @SerialName("path") val path: String
)
